src/frontend/screens/Settings/MapSettings/BackgroundMapInfo.tsx

```diff
@@ -49,77 +61,77 @@ const m = defineMessages({
-    defaultMessage: "Local highway and crossing details",
-  },
-  lvlOfDetail20: {
-    id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail20",
-    defaultMessage: "A mid-sized building",
-  },
+  // lvlOfDetail0to1: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail0to1",
+  //   defaultMessage: "Whole world",
+  // },
+  // lvlOfDetail2: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail2",
+  //   defaultMessage: "Subcontinental area ",
+  // },
+  // lvlOfDetail3to4: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail3to4",
+  //   defaultMessage: "Largest country",
+  // },
+  // lvlOfDetail5: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail5",
+  //   defaultMessage: "Large African country",
+  // },
+  // lvlOfDetail6: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail6",
+  //   defaultMessage: "Large European country",
+  // },
+  // lvlOfDetail7to8: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail7to8",
+  //   defaultMessage: "Small country, US state",
+  // },
+  // lvlOfDetail9: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail9",
+  //   defaultMessage: "Wide area, large metropolitan area",
+  // },
+  // lvlOfDetail10: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail10",
+  //   defaultMessage: "Metropolitan area",
+  // },
+  // lvlOfDetail11: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail11",
+  //   defaultMessage: "City",
+  // },
+  // lvlOfDetail12: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail12",
+  //   defaultMessage: "Town, or city district",
+  // },
+  // lvlOfDetail13to14: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail13to14",
+  //   defaultMessage: "Village, or suburb",
+  // },
+  // lvlOfDetail15: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail15",
+  //   defaultMessage: "Small road",
+  // },
+  // lvlOfDetail16: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail16",
+  //   defaultMessage: "Street",
+  // },
+  // lvlOfDetail17: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail17",
+  //   defaultMessage: "Block, park, addresses",
+  // },
+  // lvlOfDetail18: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail18",
+  //   defaultMessage: "Some buildings, trees",
+  // },
+  // lvlOfDetail19: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail19",
+  //   defaultMessage: "Local highway and crossing details",
+  // },
+  // lvlOfDetail20: {
+  //   id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail20",
+  //   defaultMessage: "A mid-sized building",
+  // },
 });
 
-type Zoom = "loading" | number | null;
+type Zoom = (number | undefined)[] | "loading";
```

Comment:
think adding `"loading"` as a type here (which is then used in React state) is kind of an odd pattern. the loading can probably be derived based on a value that is more uniform, such as `[number, number] | null`

Reply:
because it is an api call, I want to add a loading state AND the array can be `[undefined|undefined]` if there is no min/max zoom. if the array was [undefined, undefined], it would indicate the api was called and resolved (so no more loading state). 

I can replace `loading` with `null`. But the type would still have to be `(number | undefined)[] | null` to capture that undefined but non loading state

Reply:
yeah i see what you mean, although I'm wondering if when `[undefined, undefined]` would come up given that the [TileJSON spec](https://github.com/mapbox/tilejson-spec/tree/master/3.0.0#312-maxzoom) states that if the minzoom and/or maxzoom fields aren't provided, then there are default values:

- minzoom default = 0
- maxzoom default = 30

given that, i guess `[undefined, undefined]` could be a result of an error in the API call? but if that's the case, then maybe a more explicit state variable for errors would be handy

Reply:
Does optional with a default value imply that it can be undefined, but has a default value, or if it is undefined it will default to said value?

The typescript typing implies that minzoom and maxzoom can be undefined, so that would need to be addressed on the mapserver side. Given the vagueness of the `default value` and `undefined`, I wasn't treating it like an error, but I can see why this could be an error.

Reply:
yeah the more im reading the spec, the more it's clear to me that the server DOES NOT need to provide the default value for optional fields (otherwise it's not an optional field by definition). the default values are for consumers/clients to adhere to

Reply:
if thats the case, then I think keeping it at `(undefined|number)[]` makes the most sense. Do you think we should change `loading` to `null`? I personally like that it is explicitly a loading state, but I do not feel strongly about it either way

Reply:
would lean towards using null in this case given that it's a more common pattern and general consistency (i don't think there are other places where we mix primitive types like strings and arrays).

for the array, if you're keeping the `[undefined, undefined]` possibility, i would suggest using tuple types i.e. `[number, number] | [undefined, undefined]`, since the range shouldn't be more or less than two elements in length

Reply:
it could also be `[number, undefined] | [undefined, number] `. Aka max zoom could be present, but non min zoom and reverse. unless we want to define all 4 tuples, `(number|undefined)[]` feels like the most elegant way

Reply:
perhaps i misunderstood, but the values can be undefined? Or are you saying that if it is undefined, we should change it to the default values?

Reply:
> Or are you saying that if it is undefined, we should change it to the default values?

yeah this is what i was suggesting. so after traversing the tilesets to find out their min/max zooms, you may end up with one of the following, for example:

- `[undefined, 20]`. Update to `[0, 20]`
- `[10, undefined]`. Update to `[10, 30]`
- `[undefined, undefined]`. Update to `[0, 30]`

and set those converted values to the state, so that the rendering doesn't have to deal with undefined values at all

Reply:
I don't think that makes sense in terms of what we are showing to the user. If I understand correctly, the min zoom and max zoom are used to indicate when to overzoom and when to stop fetching tiles. For example, if the zoom 12 is the max zoom, and the user tries to zoom to 13, it will just overzoom 12 instead of producing more tiles. therefore having default values of 0 and 30 make sense in telling the renderer when to overzoom/not fetch tiles, since those are pretty encompassing values on either side of the spectrum, and it would endlessly try to fetch tiles if there were no values.

In this case, we are simply letting the user know what zoom level they have downloaded. So in the case that it is undefined, we do not actually know if they have 0 or 30, so I think it would be misleading to default to those values.

Reply:
Right, that is a good point about the range. Ill just show max zoom and min zoom, NOT as a range, and not show anything if it is undefined.

Reply:
Oh, no, i just re-read your comment, and I misunderstood. That is a good point about the zoom ranges. I suppose for now, we will just have to work with what we have and we can reevaluate whether or not that is actually useful info to the user. Aka, we can see if that is misleading more often than not

Reply:
yeah agreed, this is all experimental so let's get feedback (and maybe specifically ask for it?) and see what the best way to communicate that information is, if at all. all of this is a lot easier with fully online maps 😄

Reply:
So, we were definitely thinking about the min and max zoom of the downloaded tiles. I don't think `minzoom` and `maxzoom` from the tilejson spec will be helpful for a Mapeo end user, even for online maps I don't think. Don't think there is anything in tilejson that specifies the extent of available tiles in a tileset. Not sure about the architecture of mbtiles in SQLite, but for an xyz tileset in directory format, you could count the number of directories in the {z} position. Could the same be possible for iterating through an mbtiles database to get these numbers?

Reply:
got it. yeah in that case, much harder to figure out 😄 

> Could the same be possible for iterating through an mbtiles database to get these numbers?

yep! that's what we do [here](https://github.com/digidem/mapeo-map-server/blob/fa26e0a24e424fbc6f4fb5e739af8663902ddfb7/src/lib/mbtiles_import_worker.js#L145), so we could store that as some kind of metadata for the tileset and surface as a non-standardized field for the tilejson served for that tileset. this wouldn't be as sufficient for online-capable maps (because you can continuously download tiles after import), which we're not really dealing with for now so not as much of an immediate concern

Reply:
@rudokemper thanks for the insight. So would that only expose the max zoom if we did that?

Reply:
True. Rather than doing a count, guess you could also just take the lowest and highest number of the {z} directories. (Again with the same caveat that I'm not sure how the mbtiles architecture differs from xyz directory).

Reply:
yeah we'd take the min and max values of `z` found when iterating, as opposed to counting anything

Reply:
@achou11 is this something we can expose this release? Or would you rather we got rid of this for now, and add it to a subsequent release?

Reply:
i'd lean towards adding in a subsequent release just based on time constraints. i also have https://github.com/digidem/mapeo-map-server/pull/82 which would be nice to include in a new release of the map server too

Reply:
we may even be able to do this at runtime by extracting the quadkeys from the tiles that style references and taking the min+max of the result. Only concern would be potential performance penalties (i don't think quadkeys are sorted, but could be wrong??)

Reply:
added an issue for the map server here: https://github.com/digidem/mapeo-map-server/issues/83

Reply:
I have removed this, and I will add this as a ticket to do in the future